src/pathfinding_service/model/token_network.py

```diff
@@ -458,6 +459,40 @@ def check_path_request_errors(
 
         return None
 
+    def save_graph(
+        self,
+        graph_dir: Optional[str],
+        address_to_reachability: Optional[Dict[Address, AddressReachability]],
+    ) -> None:
+        """ Save a graphviz dot format description of the graph into `graph_dir`.
+
+        The graph includes capacity and (if given) reachability information.
+        Unreachable nodes will be drawn with a gray background.
+        """
+        if not graph_dir:
+            return
+        with open(os.path.join(graph_dir, datetime.utcnow().isoformat() + ".dot"), "w") as f:
+            f.write("digraph {")
+            if address_to_reachability is not None:
+                for node in self.G.nodes():
+                    f.write(
+                        "{} [{}]\n".format(
+                            to_checksum_address(node),
+                            ""
+                            if address_to_reachability[node] == AddressReachability.REACHABLE
+                            else "style=filled",
+                        )
+                    )
+            for edge_attr in self.G.edges.values():
+                f.write(
+                    '"{}" -> "{}" [label={}]\n'.format(
```

Comment:
Does graphviz handle bidirectional graphs?

Reply:
I've modeled it as a directed graph with a pair of edges between the nodes and that works.